### PR TITLE
Adding MSRE SAM model and documentation

### PR DESCRIPTION
**Pull request by hapfang**
This PR includes the SAM system model of the MSRE primary loop and the related documentation. The input file has been processed to remove the trailing whitespaces following the Moosedocs guidelines.

### CODE REVIEW

**Comment by hapfang** [comment]
Hi @GiudGiud, it looks like I couldn't add reviewers or assignees, which means I am not a project collaborator. Could you please help add the reviewers (yourself and me)? I don't know how to enable the precheck and test, can I just add moosetest as a developer in my forked repo? Thanks.

**Comment by GiudGiud** [comment]
I added you to the repo and tests should run on future pushes.

**Comment by hapfang** [comment]
It shows the tests didn't run due to failed dependencies. Do you have any thoughts? @GiudGiud, thanks!

**Comment by GiudGiud** [comment]
looks like you need to add an issue and link to it in the commit message

**Comment by hapfang** [comment]
Shall I close this PR, open an issue, and then submit a new PR with the related issue?

**Comment by GiudGiud** [comment]
no just open an issue then use git commit --amend to add the issue number to the commit

**Comment by hapfang** [comment]
@GiudGiud how could I know where and how the test failed? I can confirm that my documentation works fine locally and my SAM model works as well.

**Comment by GiudGiud** [comment]
can you click on the test results?
A bunch of VTB tests are failing right now, it may not be yours that is failing

**Comment by hapfang** [comment]
I don't know how to interpret the test results. There is too much information in the report. Could you please help double check if the error is related to the MSRE model I added?

**Comment by GiudGiud** [comment]
in fact your input is not tested. Could you please add a tests file in the msre folder.
We are moving towards regression tests so if your case is not slow let s have a CSVDiff test

**Comment by GiudGiud** [comment]
so use CSV output in the sam input file, then add a gold folder and place the standard output of this run into the folder

**Comment by hapfang** [comment]
Any example in the repo that I can follow?

**Comment by GiudGiud** [comment]
SAM should have plenty of CSVdiff tests. In the repo we only have syntax tests so far. 
You can look at the tests file in msfr/steady for inspiration

**Comment by hapfang** [comment]
The SAM case takes about 30 min to reach the steady state. I have added the regression test with the reference CSV dataset.

**Comment by GiudGiud** [comment]
ok that is much too long. 
Let s use the `cli_args` field in the test specification to compare it after like 10 time steps only

**Comment by hapfang** [comment]
Could you please share an example of `cli_args`, or help edit the tests script? I can run the case for 10 steps to generate the reference data. Thanks.

**Comment by GiudGiud** [comment]
See this file where I did just that
https://github.com/idaholab/virtual_test_bed/blob/main/pbfhr/sam_model/tests

**Comment by hapfang** [comment]
Thanks @GiudGiud , I have made the changes accordingly in tests, and uploaded a new reference dataset.

**Comment by GiudGiud** [comment]
Great thanks!
I have a tight deadline on a report, I will get to reviewing this PR this weekend.

**Review by GiudGiud — APPROVED** [review]

File: msre/tests [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[Tests]
+  [msre_sam]
+    type = 'CSVdiff'
```
Comment (GiudGiud) [inline msre/tests:3]:
```suggestion
    type = 'CSVDiff'
```

**Comment by GiudGiud** [comment]
The test failure is simply that doesnt allow warnings in tests. The allow_warnings flag is being overriden in SAM source.
Just need to comment this line in run_tests `sys.argv.insert(1, "--error")`
do you make PRs to SAM? I could do it too

**Comment by hapfang** [comment]
Thanks a lot for the update Guillaume. I have not worked with the SAM source code directly. If it is something worthwhile to update, please help make a PR and I am sure the SAM team will appreciate it!

File: msre/tests [hunk 2]
```diff
@@ -0,0 +1,9 @@
+[Tests]
+  [msre_sam]
+    type = 'CSVDiff'
+    input = 'msre_loop_1d.i'
+    csvdiff = 'msre_loop_1d_csv.csv'
+    cli_args = 'Executioner/num_steps=10'
+    allow_warnings = true
+  []
```
Comment (GiudGiud) [inline msre/tests:8]:
```suggestion
    executable_pattern = 'sam*|blue_crab*'
  []
```

Comment (hapfang) [reply msre/tests:9]: The suggested change looks good to me.

**Comment by GiudGiud** [comment]
SAM PR got merged. Once it propagates this PR should pass tests

**Comment by hapfang** [comment]
Sounds great. Thanks Guillaume.

**Comment by GiudGiud** [comment]
Seems blue crab struggles with this input. I ll have a look

**Comment by hapfang** [comment]
OK, thanks.

**Comment by hapfang** [comment]
Thanks a lot for the update.

**Comment by GiudGiud** [comment]
Test failures are unrelated and being addressed in #27

**Comment by GiudGiud** [comment]
All good
Thanks for the contribution and the patience Jun!

We may move this from msre/ to msr/msre in the future btw.

**Comment by hapfang** [comment]
Sounds good. Thank you for all the help Guillaume! I appreciate it.

**Pull request merged**